web/src/message_view.js

```diff
@@ -1065,10 +1066,9 @@ export function render_message_list_with_selected_message(opts) {
     narrow_history.save_narrow_state_and_flush();
 }
 
-export function activate_stream_for_cycle_hotkey(stream_id) {
-    const stream_name = stream_data.get_stream_name_from_id(stream_id);
+function activate_stream_for_cycle_hotkey(stream_id) {
     // This is the common code for A/D hotkeys.
-    const filter_expr = [{operator: "channel", operand: stream_name}];
+    const filter_expr = [{operator: "channel", operand: stream_id.toString()}];
```

Comment:
I feel like a lot of code paths calling `show` .

Would it make sense to have `operand` being an integer be a valid encoding?

```$ git grep operand | grep toString
web/src/filter.ts:            return message.id.toString() === operand;
web/src/filter.ts:                operand = operand.toString().toLowerCase();
web/src/filter.ts:                operand = operand.toString().toLowerCase();
web/src/filter.ts:                operand = operand.toString().toLowerCase();
web/src/filter.ts:            return sign + operator + ":" + Filter.encodeOperand(term.operand.toString());
```

Reply:
Did you mean to say "I feel like a lot of code paths call `toString`"?

I meant to talk a bit about this, as I was thinking about it. I agree converting between string and integer in several places is not great.

However, it feels a bit hard to add an integer operand, because `channel:Denmar` is a parseable search term, and we only change the channel operand to a number once the stream name is valid -- so channel can have string and (currently stringified) number operands. I don't really like that this is the case, and maybe there's a nicer way to parse terms that's more elegant, but it feels like supporting integer channel id operands would be a much bigger change.

Reply:
Yeah, that's what I meant.

Note that `filter.parse` has to handle a raw string, but there's nothing requiring the fact that you can type `"channel:Denmark" to say anything about the required format of processed `NarrowTerm` structures that we use; it would be totally reasonable to have parsing turn channel names into either an ID or an error, without that needing to contaminate these internal APIs).

Reply:
`search_suggestion.ts` uses `NarrowTerm`s that we parsed, and that has to include partially typed out channel names which are part of how we give channel suggestions.

Reply:
If there's only one place that needs a different format, that code path could do whatever logic is required to produce correct output; I don't see why that should have a strong constraint on the format of internal data structures.

Reply:
This seems worth considering, but as a followup. I think it would involve separating `NarrowTerm` into `SearchTerm` (all string) and `NarrowTerm` (only after validation, where channels always have number operands).

Reply:
Yeah, that seems like a reasonable way to proceed here.

Reply:
Cool, added this to my list of non-urgent and somewhat-involved followups